api/src/search/backend/load_opportunities_to_index.py

```diff
@@ -120,71 +124,74 @@ def incremental_updates_and_deletes(self) -> None:
-            self.db_session.query(JobLog)
-            .filter(
-                JobLog.job_type == self.cls_name(),
-                JobLog.job_status == JobStatus.COMPLETED,
-            )
-            .order_by(JobLog.created_at.desc())
-            .first()
-        )
-
-        # Fetch opportunities that need processing from the queue
-        query = (
+    def _build_opportunities_to_process_query(self) -> Select:
+        return (
             select(Opportunity)
             .join(OpportunityChangeAudit)
             .join(CurrentOpportunitySummary)
             .where(
                 Opportunity.is_draft.is_(False),
                 CurrentOpportunitySummary.opportunity_status.isnot(None),
+                OpportunityChangeAudit.is_loaded_to_search.isnot(True),
             )
+            .order_by(Opportunity.created_at.desc())
             .options(selectinload("*"), noload(Opportunity.all_opportunity_summaries))
+            .limit(self.config.incremental_load_batch_size)
         )
 
-        # Add timestamp filter
-        if last_successful_job:
-            query = query.where(OpportunityChangeAudit.updated_at > last_successful_job.created_at)
-
-        queued_opportunities = self.db_session.execute(query).scalars().all()
+    def _handle_incremental_upserts(self, existing_opportunity_ids: set[int]) -> None:
+        """Handle updates/inserts of opportunities into the search index when running incrementally"""
+        while True:
+            # Check elapsed_time before starting new batch processing
+            elapsed_time = utcnow() - self.start_time
 
-        # Process updates and inserts
-        processed_opportunity_ids = set()
-        opportunities_to_index = []
+            if elapsed_time.total_seconds() > self.config.incremental_load_max_process_time:
+                logger.info(
+                    f"Elapsed time: {elapsed_time.total_seconds() / 60:.2f} minutes exceeded the limit. Stopping batch processing."
+                )
+                break
 
-        for opportunity in queued_opportunities:
-            logger.info(
-                "Processing queued opportunity",
-                extra={
-                    "opportunity_id": opportunity.opportunity_id,
-                    "status": (
-                        "update"
-                        if opportunity.opportunity_id in existing_opportunity_ids
-                        else "insert"
-                    ),
-                },
+            # Fetch opportunities that need processing from the queue
+            queued_opportunities = (
+                self.db_session.execute(self._build_opportunities_to_process_query())
+                .scalars()
+                .all()
             )
 
-            # Add to index batch if it's indexable
-            opportunities_to_index.append(opportunity)
-            processed_opportunity_ids.add(opportunity.opportunity_id)
+            if len(queued_opportunities) == 0:
+                break
```

Comment:
Could we add a message sorta like this?
```suggestion
            if len(queued_opportunities) == 0:
                logger.info("No opportunities remain in queue, exiting processing")
                break
```